File: code_samples/python/v1.0.0/confirm_cancel_pre_authorization.py
from tatrapayplus import TBPlusSDK
from tatrapayplus.models import *

client = TBPlusSDK(
    "your-client-id",
    "your-client-secret"
)
payment_id = "b54afd37-5bb9-4080-9416-5ec450779087"
cancel_pre_authorization_data = CardPayUpdateInstruction(
    operation_type=CardPayUpdateInstructionOperationType.CANCEL_PRE_AUTHORIZATION,
)
chargeback_response = client.update_payment(payment_id, cancel_pre_authorization_data)

# or

cancel_pre_authorization_data = CardPayUpdateInstruction(
    operation_type=CardPayUpdateInstructionOperationType.CONFIRM_PRE_AUTHORIZATION,
)
chargeback_response = client.update_payment(payment_id, cancel_pre_authorization_data)

File: code_samples/php/v1.0.0/confirm_cancel_pre_authorization.php
<?php
use Tatrapayplus\TatrapayplusApiClient\Api\TatraPayPlusAPIApi;
use Tatrapayplus\TatrapayplusApiClient\Model\CardPayUpdateInstruction;

$tatrapayplus_api = new TatraPayPlusAPIApi(
    "your-client-id",
    "your-client-secret",
);
$payment_id = 'b54afd37-5bb9-4080-9416-5ec450779087'; // Retrieved from initiatePayment

$pre_authorization_data = new CardPayUpdateInstruction([
    "operation_type" => CardPayUpdateInstruction::OPERATION_TYPE_CANCEL_PRE_AUTHORIZATION,
]);
$response = $tatrapayplus_api->updatePaymentIntent($payment_id, $pre_authorization_data);

// or

$pre_authorization_data = new CardPayUpdateInstruction([
    "operation_type" => CardPayUpdateInstruction::OPERATION_TYPE_CONFIRM_PRE_AUTHORIZATION,
]);
$response = $tatrapayplus_api->updatePaymentIntent($payment_id, $pre_authorization_data);